### PR TITLE
libiconv: Just include the iconv headers

### LINKED ISSUE #19895: Libiconv in build inputs leads to subtly broken cflags on Linux
`glibc` includes `iconv` functions so we used to have `libiconvOrNull` to avoid a spurious build input on systems that have `glibc` as their libc. We removed this in #6166 (for good reason, IMO) and now just have `libiconv` as an alias for `glibc` on those systems. Unfortunately, including `glibc` as a build input puts its include headers on the search path with `isystem` instead of just using the `idirafter` baked into the cc wrapper, which means glibc's headers are no longer searched last in the list as desired. This leads to the bug referenced in #19834 (see discussion on #19859 ), and in general will cause problems for anything that builds with libc++, depends on libiconv, and references `<cmath>` on glibc systems.

I'm not sure the best solution here. We could ignore instances of the c library our compiler is built with in build inputs, we could have some logic in glibc's setup hook to tell it to add `idirafter` instead of `isystem`, we could bring back `libiconvOrNull` or something like it... None of these seem obviously good or obviously awful. Thoughts?

### PR DESCRIPTION
**Pull request by shlevy**
Fixes #19895

### CODE REVIEW

**Comment by shlevy** [comment]
http://hydra.nixos.org/jobset/nixpkgs/glibc-iconv-19919

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -7937,10 +7937,18 @@ in
     (if crossSystem.libc == "glibc" then libcCross
```
Comment (Ericson2314) [inline pkgs/top-level/all-packages.nix]: Any reason `glibcIconv` isn't also applied to this?

CC @DavidEGrayson

Comment (shlevy) [inline pkgs/top-level/all-packages.nix]: No reason except lack of time/need on my part.

Comment (Ericson2314) [inline pkgs/top-level/all-packages.nix]: That's fine; just checking.

**Pull request merged**